modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/FileUtilWrapper.kt

```diff
@@ -1,9 +1,38 @@
 package au.com.shiftyjelly.pocketcasts.utils
 
+import android.content.Context
+import android.graphics.Bitmap
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import timber.log.Timber
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
 import javax.inject.Inject
 
 class FileUtilWrapper @Inject constructor() {
     fun deleteDirectoryContents(path: String) {
         FileUtil.deleteDirectoryContents(path)
     }
+
+    suspend fun saveBitmapToFile(
+        bitmap: Bitmap,
+        context: Context,
+        saveFolderName: String,
+        saveFileName: String,
+    ): File? = withContext(Dispatchers.IO) {
+        val imagesFolder = File(context.cacheDir, saveFolderName)
+        var file: File? = null
+        try {
+            imagesFolder.mkdirs()
+            file = File(imagesFolder, saveFileName)
+            val stream = FileOutputStream(file)
+            bitmap.compress(Bitmap.CompressFormat.PNG, 90, stream)
+            stream.flush()
+            stream.close()
```

Comment:
I think [`use`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html) may be a more robust way to ensure the file stream gets closed.

```suggestion kotlin
            FileOutputStream(file).use { stream ->
                bitmap.compress(Bitmap.CompressFormat.PNG, 90, stream)
                stream.flush()
            }
```